cumulusci/core/keychain/tests/test_encrypted_file_project_keychain.py

```diff
@@ -188,17 +259,117 @@ def test_load_service_files(self, keychain):
+        service_config = ServiceConfig(
+            {"name": "foo", "password": "1234", "token": "1234"}
+        )
+        with env:
+            env.set(f"{service_prefix}github", json.dumps(service_config.config))
+            env.set(f"{service_prefix}github", json.dumps(service_config.config))
+            keychain._load_services_from_environment()
+
+        assert 1 == 1
+
+    def test_get_service__built_in_connected_app(self, keychain):
+        built_in_connected_app = keychain.get_service("connected_app")
+        assert built_in_connected_app is DEFAULT_CONNECTED_APP
+
+    def test_get_service__with_class_path(self, keychain, service_config):
+        encrypted = keychain._encrypt_config(service_config)
+        keychain.config["services"]["marketing_cloud"] = {"foo": encrypted}
+        mc_service = keychain._get_service("marketing_cloud", "foo")
 
+        assert isinstance(mc_service, MarketingCloudServiceConfig)
+
+    @mock.patch("cumulusci.core.keychain.encrypted_file_project_keychain.import_class")
+    def test_get_service__bad_class_path(self, import_class, keychain, service_config):
+        import_class.side_effect = AttributeError
+        encrypted = keychain._encrypt_config(service_config)
+        keychain.config["services"]["marketing_cloud"] = {"foo": encrypted}
+
+        error_message = "Unrecognized class_path for service: cumulusci.core.config.marketing_cloud_service_config.MarketingCloudServiceConfig"
+        with pytest.raises(CumulusCIException, match=error_message):
+            keychain._get_service("marketing_cloud", "foo")
+
+    def test_get_service__does_not_exist(self, keychain, service_config):
+        keychain.set_service("github", "alias", service_config)
+        error_message = "No service of type github exists with the name: does-not-exist"
+        with pytest.raises(ServiceNotConfigured, match=error_message):
+            keychain.get_service("github", "does-not-exist")
+
+    @pytest.mark.skipif(
+        sys.platform.startswith("win"),
+        reason="Windows has a different value returned by os.stat",
+    )
     def test_set_service_github(self, keychain, service_config):
         keychain.set_service("github", "alias", service_config)
         default_github_service = keychain.get_service("github")
 
+        service_filepath = Path(
+            keychain.global_config_dir, "services/github/alias.service"
+        )
+        assert service_filepath.is_file()
+
+        # ensure expected file permissions
+        stat_result = os.stat(service_filepath)
+        actual_mode = oct(stat_result.st_mode & 0o777)
+        assert actual_mode == oct(SERVICE_ORG_FILE_MODE)
+
         assert default_github_service.config == {
             **service_config.config,
             "token": "test123",
         }
 
+    @pytest.mark.skipif(
+        not sys.platform.startswith("win"),
+        reason="Windows has a different value returned by os.stat",
+    )
```

Comment:
Ditto

Reply:
This test and the one above are the same thing, but one is for windows, and the other is not.